deker/validators.py

```diff
@@ -42,17 +44,37 @@ def process_time_dimension_attrs(attributes: dict, attr_name: str) -> datetime.d
     return time_attribute
 
 
+def __validate_attribute_type(attribute: object) -> None:
+    """Validate attribute type over allowed types.
+
+    :param attribute: attribute object
+    """
+    if isinstance(attribute, tuple):
+        for attr in attribute:
+            __validate_attribute_type(attr)
+
+    dtype = type(attribute)
+    try:
+        DTypeEnum(dtype).value
+    except (ValueError, KeyError):
+        raise DekerValidationError(f"Invalid dtype value {dtype}")
+
+
 def __process_attributes_types(
     attrs_schema: Tuple["AttributeSchema", ...],
     primary_attributes: dict,
     custom_attributes: dict,
+    dimensions: list,
 ) -> None:
     """Validate attributes types over schema and update dicts if needed.
 
     :param attrs_schema: attributes schema
     :param primary_attributes: primary attributes to validate
     :param custom_attributes: custom attributes to validate
+    :param dimensions: list of dimensions
     """
+    from deker.schemas import TimeDimensionSchema
```

Comment:
May be it makes sense to move it to the branch where you deal with TimeDimension ?

Reply:
I'm not sure, I got what you mean.
We shall check if any TDS's `start_value` has a reference to an attribute schema. If you look at imports you'll see, that all schemas are imported just for type checking. I can check if normal import makes problems.

Reply:
Yep, I think it's better to move either to all imports, or somewhere near 105 line. Or we can leave it as is, not a big deal